### ISSUE TITLE
nsd does not log to anywhere

### ISSUE THREAD

**Issue by chris13524** [mail-in-a-box/mailinabox#1239]
`logfile: "/var/log/nsd.log"` should be added under the `server` heading in `/etc/nsd/nsd/conf` in order to preserve logs.

**Comment by nikksno** [comment]
+1. Also I think once we have this log running we should check such logfile to display any errors in "system checks", as that will cause nsd to stop resolving new records in that zone. @JoshData I know you mean to find a better way to validate records and / or check that nsd accepts them, but an other-way-around check to see if nsd throws any errors would also be helpful, I think, as we'll never actually be able to be 100% sure that nsd is operating correctly otherwise. Don't you think?

**Comment by JoshData** [comment]
> check to see if nsd throws any errors would also be helpful

For sure, if we can validate with nsd itself that would be great.

**Comment by nikksno** [comment]
Yes it would, but as a double check, can we start by also looking at the nsd error log for any error [or any uncaught error in the future when we actually start validating against nsd]?

**Comment by JoshData** [comment]
Sure but I think practically speaking that will be difficult to do correctly, and once nsd goes down you run the risk that the control panel won't be accessible after that anyway.

**Comment by chris13524** [comment]
In that case, you log into the server and check the error logs!

**Comment by nikksno** [comment]
@JoshData it would via IP address, no? And still, nsd going down completely is an edge case, I've never gotten to that point, but without changing the config to add the logfile and reading it I'd have never gotten to the end of the various problems. In any event, I would start by adding the logfile definition in the nsd config, and then we can start a list of improvements to be done to either validate records better in MiaB or to validate them against nsd directly, don't you think?

**Comment by JoshData** [comment]
I already said sure. :)
